mainwindow.cpp

```diff
@@ -309,10 +310,27 @@ namespace dealii
           return false;
         };
 
-      XMLParameterWriter  xml_writer(tree_widget);				// create a xml_writer
+      if (filename.contains(".xml"))
+        {
+          XMLParameterWriter writer(tree_widget);                               // create a xml writer
+          if (!writer.write_xml_file(&file))                                    // and write the xml file
+            return false;
+        }
+      else if (filename.contains(".prm"))
+        {
+          PRMParameterWriter  writer(tree_widget);                              // create a prm writer
+          if (!writer.write_prm_file(&file))                                    // and write the prm file
+            return false;
+        }
+      else
+        {
+          QMessageBox::warning(this, tr("parameterGUI"),
+                                     tr("Unknown output format: %1.")
+                                     .arg(filename));
+          file.remove();
```

Comment:
Yes. The documentation says it is closed before removing the file (and I checked that it works on my machine).